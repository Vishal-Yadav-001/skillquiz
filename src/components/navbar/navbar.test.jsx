import { act, fireEvent, render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Navbar from "./navbar.component";
import renderer from "react-test-renderer";

describe("Navbar", () => {
  test("should display Navbar", () => {
    const navbar = renderScreen();
    screen.debug();
    expect(screen.getByText("SkillScore").innerHTML).toBe("SkillScore");
  });
  test("should display login button when user is new", () => {
    localStorage.clear();
    const navbar = renderScreen();
    expect(screen.getByText("Login")).toBeInTheDocument();
  });
  test("should display logout button when user is has registered", () => {
    const navbar = render(<Navbar isLogined={true} />, {
      wrapper: BrowserRouter,
    });
    expect(screen.getByText("Logout")).toBeInTheDocument();
  });
  test("should  logout  when user choose to logout", () => {
    const navbar = render(
      <Navbar
        navbarHeight={navbarHeight}
        handleDrawerToggle={handleDrawerToggle}
        setIsLogined={setIsLogined}
        user={user}
        isLogined={true}
      />,
      {
        wrapper: BrowserRouter,
      }
    );
    const button = screen.getAllByText("Logout")[0];
    act(() => {
      fireEvent.click(button);
    });
    expect(localStorage.getItem("User")).toBeNull();
  });

  test("snapshot", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
        <Navbar
          navbarHeight={navbarHeight}
          handleDrawerToggle={handleDrawerToggle}
          setIsLogined={setIsLogined}
          user={user}
          isLogined={true}
        />
        </BrowserRouter>
       
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

const renderScreen = () => {
  return render(<Navbar />, { wrapper: BrowserRouter });
};

const user = { name: "gfg", email: "cccc@gmail.com", password: "ccc" };
const navbarHeight = 65;
const handleDrawerToggle = () => {
  return null;
};
const setIsLogined = () => {
  return null;
};
