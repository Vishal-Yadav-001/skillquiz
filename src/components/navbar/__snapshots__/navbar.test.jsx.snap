// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar snapshot 1`] = `
<div
  className="MuiBox-root css-i9gxme"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-1ljqnpo-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      <button
        aria-label="menu"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-ancrnh-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </button>
      <div
        className="MuiTypography-root MuiTypography-h6 css-195gxv0-MuiTypography-root"
      >
        SkillScore
      </div>
      <div
        className="MuiAvatar-root MuiAvatar-circular css-fx535s-MuiAvatar-root"
      >
        <img
          alt="User Name"
          className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
          src="young-man.png"
        />
      </div>
      <p
        className="MuiTypography-root MuiTypography-body1 css-1apgjqp-MuiTypography-root"
      >
        gfg
      </p>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-103i3hd-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Logout
      </button>
    </div>
  </header>
</div>
`;
